src/index.ts

```diff
@@ -101,6 +101,33 @@ const getScalarDefinition = (value: ScalarDefinition | ScalarGeneratorName): Sca
     return value;
 };
 
+const getCustomScalarValue = (customScalar: ScalarDefinition, opts: Options<NamedTypeNode>) => {
+    // If there is a mapping to a `casual` type, then use it and make sure
+    // to call it if it's a function
+    const embeddedGenerator = casual[customScalar.generator];
+    if (!embeddedGenerator && customScalar.generator) {
+        return customScalar.generator;
+    }
+
+    const generatorArgs: unknown[] = Array.isArray(customScalar.arguments)
+        ? customScalar.arguments
+        : [customScalar.arguments];
+    if (opts.dynamicValues) {
+        return `casual['${customScalar.generator}']${
+            typeof embeddedGenerator === 'function' ? `(...${JSON.stringify(generatorArgs)})` : ''
+        }`;
+    }
+    const value = typeof embeddedGenerator === 'function' ? embeddedGenerator(...generatorArgs) : embeddedGenerator;
+
+    if (typeof value === 'string') {
+        return `'${value}'`;
+    }
+    if (typeof value === 'object') {
+        return `${JSON.stringify(value)}`;
+    }
+    return value;
+};
```

Comment:
From the viewpoint of increasing code simplicity, the mock generation of customScalar could also be moved to the MockValueGenerator class. This eliminates the need to import casual in index.ts.